src/Data/ElfEdit/HighLevel/Layout.hs

```diff
@@ -57,7 +57,7 @@ updateSegments fn = elfFileData (updateSeq impl)
       let inner = updateSeq impl (elfSegmentData seg)
           updateData s d = s { elfSegmentData = d }
           newSeg :: f (Maybe (ElfSegment w))
-          newSeg = fn =<< (fmap (updateData seg) inner)
+          newSeg = (fn . updateData seg) =<< inner
```

Comment:
If _I_ were writing this, I'd opt to be explicit and just write:

```
newSeg = do x <- inner
            fn (updateData seg x)
```

But the way it's currently written is fine too. (Similarly for the `(.)`/`(=<<)` combo on line 79.)